### ISSUE TITLE
`estimated_due_date` calculator implementation details

### ISSUE THREAD

**Issue by Nozom1466** [ncbi-nlp/MedCalc-Bench#7]
Thanks for your great work! 

I have an issue when I check [estimated_due_date](https://github.com/ncbi-nlp/MedCalc-Bench/blob/main/calculator_implementations/estimated_due_date.py) calculator implementation: 

https://github.com/ncbi-nlp/MedCalc-Bench/blob/0d3fba6dacc8ab61d886a495b7a0a55204b0219d/calculator_implementations/estimated_due_date.py#L20-L27

I am not sure if **adding** `cycle_length_gap` in Line 22 is correct. It seems that `cycle_length_gap` should be **subtracted** given `explanation` in Line 23. 

And a corresponding example in your generated dataset is as follows:

```json
{
    "Row Number": "9654",
    "Calculator ID": "13",
    "Calculator Name": "Estimated Due Date",
    "Category": "date",
    "Output Type": "date",
    "Note ID": "22",
    "Note Type": "Template",
    "Patient Note": "The patient's last menstrual period was on 09/17/2011. Her cycle length is 21 days.",
    "Question": "Using Naegele's Rule for estimated due date based on the last menstrual period and cycle length, what is the the patient's estimated due date? Your response should be in the format of M/D/Y (ie 08/31/2023, 07/03/2000) with just the date and not other text. ",
    "Relevant Entities": {
      "cycle length": 21,
      "Last menstrual date": "09/17/2011"
    },
    "Ground Truth Answer": "06/30/2012",
    "Lower Limit": "06/30/2012",
    "Upper Limit": "06/30/2012",
    "Ground Truth Explanation": "The patient's estimated due date based on their last period is computed by using Naegele's Rule. Using Naegele's Rule, we add 40 weeks to the patient's last menstrual period date. We then add or subtract days from the patient's estimated due date depending on how many more or less days a patient's cycle length is from the standard 28 days. \nThe patient's last menstrual period was 09/17/2011. \nThe date after adding 40 weeks to the patient's last menstrual period date is 06/23/2012. \nBecause the patient's cycle length is 21 days, this means that we must subtract 7 days from the patient's estimate due date. Hence, the patient's estimated due date is 06-30-2012. ",
    "Unit": "",
    "Original Relevant Entities": "{'cycle length': 21, 'Last menstrual date': '09/17/2011'}"
  },
```

Looking forward to your response!

**Comment by nikhilk7153** [comment]
Hi Ryan, yes, you are correct about this. I will update the repository accordingly. I am in the process of curating a 1.1 version which resolves a few other issues like this one (we only wrote a few test cases to verify our results for each calculator, so we could have missed out on some edge cases) and adds 53 more instances so I will update the dataset once I finish curating that. For now, I will just update implementation. Thanks!

**Comment by nikhilk7153** [comment]
Also, if

**Comment by Nozom1466** [comment]
> Hi Ryan, yes, you are correct about this. I will update the repository accordingly. I am in the process of curating a 1.1 version which resolves a few other issues like this one (we only wrote a few test cases to verify our results for each calculator, so missed out on a few instances) and adds 53 more instances so I will update the dataset once I finish curating that. For now, I will just update implementation. Thanks!

Looking forward to version 1.1!

**Comment by nikhilk7153** [comment]
Hi Ryan, for the time being, I have updated the test and train sets to the following: 


test_data: https://docs.google.com/spreadsheets/d/1qUyF6LhJt1fYij3d3KpmRAmB_7maIoBfLKd_cY7CiZs/edit?usp=sharing

train_data: https://docs.google.com/spreadsheets/d/1y36URmNF3t5xlXujjFuGdMbjyk9H9xohiSIB1Jec0oQ/edit?usp=sharing

I will update the repository soon, but just wanted to give these to you after I go through one more round of verification.

**Comment by Nozom1466** [comment]
> Hi Ryan, for the time being, I have updated the test and train sets to the following: 
> 
> 
> test_data: https://docs.google.com/spreadsheets/d/1qUyF6LhJt1fYij3d3KpmRAmB_7maIoBfLKd_cY7CiZs/edit?usp=sharing
> 
> train_data: https://docs.google.com/spreadsheets/d/1y36URmNF3t5xlXujjFuGdMbjyk9H9xohiSIB1Jec0oQ/edit?usp=sharing
> 
> I will update the repository soon, but just wanted to give these to you after I go through one more round of verification. 

Thanks for the dataset update!

**Comment by Nozom1466** [comment]
> Hi Ryan, for the time being, I have updated the test and train sets to the following:
> 
> test_data: https://docs.google.com/spreadsheets/d/1qUyF6LhJt1fYij3d3KpmRAmB_7maIoBfLKd_cY7CiZs/edit?usp=sharing
> 
> train_data: https://docs.google.com/spreadsheets/d/1y36URmNF3t5xlXujjFuGdMbjyk9H9xohiSIB1Jec0oQ/edit?usp=sharing
> 
> I will update the repository soon, but just wanted to give these to you after I go through one more round of verification.

Hello! I found possibly an unit error in Row Number 4594 "Relevant Entities" column in your updated `train_data`: `"{'weight': [1.4, 'sds']}"`, which seems to be `"{'weight': [1.4, 'kg']}"`


4594,22,Maintenance Fluids Calculations,physical,decimal,pmc-5606946-1,Extracted,"A 42-year-old man with known cystic fibrosis (CF) and secondary diabetes was referred to Endocrinology for management of his Addison’s disease and androgen replacement. He was diagnosed with CF aged 4 years. As a consequence of CF he suffered mild abrogation of growth, with a final height of −2.5 SDS. His corresponding weight was −1.4 SDS, and pubertal development was assessed to be Tanner stage 5 at age 16. At age 18 years he was diagnosed with CF-related diabetes mellitus requiring insulin. During admission with a respiratory exacerbation of his CF at age 19 years he was noted to be pigmented, with serum sodium 124 mmol/L and potassium 5.1 mmol/L. Further tests confirmed cortisol insufficiency [9am cortisol <50 nmol/L (<1.8 µg/dL)] in association with unrecordable aldosterone [<55pmol/L (<1.5 ng/dL)]. Plasma renin activity is not available from the time of diagnosis. The patient was diagnosed with Addison’s disease and commenced on hydrocortisone and fludrocortisone replacement. At the time of the AI diagnosis the patient was on a regular Budesonide inhaler. However he was not on any oral medications (i.e. oral steroids or imidazoles), which could have caused iatrogenic adrenal insufficiency.
At 35 years of age he underwent lung transplantation for end-stage pulmonary disease, necessitating long-term corticosteroid and immunosuppressive therapy with tacrolimus and mycophenolate mofetil. At age 38 years he presented with symptoms of low libido, was found to have HH [LH 1.6 IU/L, FSH 10.2 IU/L, testosterone 3.3 nmol/L (95 ng/dL), SHBG 48 nmol/L, estradiol 54pmol/L (14.7 pg/mL)], and was commenced on testosterone replacement. At the time of diagnosis of HH the patient was on hydrocortisone, as well as tacrolimus and mycophenolate mofetil. He was then lost to endocrine follow-up, however, on re-presentation aged 42 years, a detailed history revealed he had a brother diagnosed with Addison’s disease at a similar age. Investigations of additional pituitary axes were within normal limits [TSH 1.29 miu/L (reference range 0.2–4.0), free T4 17.9 pmol/L (reference range 10.0–20.0), prolactin 278 miu/L (normal level <600)], and a pituitary MRI was unremarkable. Neither the patient nor his brother have fathered children. Given the presence of Addison’s disease with HH, potential infertility, and family history, the patient was tested for mutation in the DAX-1 (NR0B1) gene. DNA analysis revealed a heterozygous c.775T > C (p.Ser259Pro) substitution within exon 1 of the DAX gene, which changes codon 259 from serine to proline in the putative LBD (Fig. ). The results of in silico analysis suggested causality. p.Ser259 is a highly conserved amino acid in the ligand binding domain of the nuclear hormone receptor core of DAX1. SIFT () predicts that p.Ser259Pro will affect protein function. Polyphen () predicts p.Ser259Pro to be probably damaging. Russell analysis () predicts the amino acid substitution from serine to proline to be disfavoured. Splicing (Alamut) c.775T > C is predicted that the variant does not affect the splicing of DAX1. The patient has declined genetic counseling, as he has not had any family plans so far.","Based on the patient's weight, what is the patient's maintenance fluid in mL/hr?  You should use the patient's medical values and health status when they were first admitted to the hospital prior to any treatment.","{'weight': [1.4, 'sds']}",5.6,5.32,5.88,"For patient's with weight less than 10 kg, the rule for computing maintenance fluid is to multiply their weight by 4 mL/kg/hr to get the maintenance fluids per hour.
For patient's with weight between 10 kg and 20 kg, the formula for computing maintenance fluid is 40 mL/hr + 2 mL/kg/hr * (weight (in kilograms) - 10 kilograms).
For patient's with weight greater than 20 kg, the formula for computing the maintenance fluid is 60 mL/hr + 1 mL/kg/hr * (weight (in kilograms) - 20 kilograms).
The patient's weight is 1.4 kg. Hence, the patient's maintenance fluid is 1.4 kg * 4 mL/kg/hr = 5.6 mL/hr.
Hence, the patient's fluid maintenance is 5.6 mL/hr."

**Comment by nikhilk7153** [comment]
Hi, you are correct that this is an issue. I can remove this instance on the excel sheet, but as mentioned in the paper, please note that we don't provide such verifications for the training set like we do for the test set due to the cost of annotations.

**Comment by nikhilk7153** [comment]
Hi Ryan, I have removed this instance from the train data csv: https://docs.google.com/spreadsheets/d/1qp9fJLiq8N6HGKJz1QeXXLZz3ZKmYWeBoTZ-IM-6fzo/edit?gid=1483105462#gid=1483105462 

Here's the most up-to-date one. While we are not going to hire doctors to verify the training dataset, if you (or anyone else) find anything that should be changed, I'm always happy to update the dataset.
